calico_containers/integrations/kubernetes/tests/kube_plugin_test.py

```diff
@@ -0,0 +1,378 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+if sys.version_info.major == 2:
+    import __builtin__ as builtins
+else:
+    import builtins
+import json
+import unittest
+import socket
+from nose.tools import assert_equal, assert_true, assert_false
+from mock import patch, Mock, MagicMock, mock_open
+from subprocess import CalledProcessError
+from calico_containers.integrations.kubernetes import calico_kubernetes
+from pycalico.datastore import IF_PREFIX
+
+class NetworkPluginTest(unittest.TestCase):
+
+    def setUp(self):
+        # Mock out sh when importing calico_kubernetes so it doesn't fail when
+        # trying to find the calicoctl binary (which may not exist)
+        with patch('calico_containers.integrations.'
+                      'kubernetes.calico_kubernetes.sh') as m_sh:
+            self.plugin = calico_kubernetes.NetworkPlugin()
+
+    def test_create(self):
+        with patch.object(self.plugin, '_configure_interface') as m_interface, \
+                patch.object(self.plugin, '_configure_profile') as m_prof:
+            # Set up mock objects
+            m_interface.return_value = 'endpt_id'
+
+            # Set up args
+            args = ['script name','setup','','pod-name','dockerId']
+
+            # Call method under test
+            self.plugin.create(args)
+
+            # Assert
+            self.assertEqual('pod_name', self.plugin.pod_name)
+            self.assertEqual('dockerId', self.plugin.docker_id)
+            m_interface.assert_called_once_with()
+            m_prof.assert_called_once_with('endpt_id')
+
+    def test_create_fail(self):
+        with patch.object(self.plugin, '_configure_interface') as m_interface, \
+                patch.object(self.plugin, '_configure_profile') as m_prof, \
+                patch('calico_containers.integrations.'
+                      'kubernetes.calico_kubernetes.sys.exit') as m_sys:
+            # Set up mock objects
+            m_interface.side_effect = CalledProcessError(1,'','')
+
+            # Set up args
+            args = ['script name','setup','','podname','dockerId']
+
+            # Call method under test
+            self.plugin.create(args)
+
+            # Assert
+            m_sys.assert_called_once_with(1)
+
+    def test_delete(self):
+        with patch.object(self.plugin, 'calicoctl') as m_calicoctl:
+            # Set up args
+            args = ['script name','teardown', '','pod-name','dockerId']
+
+            # Call method under test
+            self.plugin.delete(args)
+
+            # Assert
+            self.assertEqual(self.plugin.pod_name, 'pod_name')
+            self.assertEqual(self.plugin.docker_id, 'dockerId')
+            m_calicoctl.assert_any_call('container', 'remove', 'dockerId')
```

Comment:
Could you replace "assert_any_call" with "assert_has_calls" - it has the benefit of asserting that they are called in the correct order.